Design/Fit functions in python.md

```diff
@@ -0,0 +1,198 @@
+# Working with fit functions in python
+
+Although fitting functions have some of the functionality exposed to python the use of it is very limited and functions are constructed via strings. Values of the optimised parameters get extracted from `TableWorkspace`s output from the Fit algorithm. Both function construction and extraction of the results are cumbersome and error prone. This document describes a design of a python fitting API aiming to overcome these problems. The main idea of the solution proposed here is to create and manipulate a C++ [`IFunction`](https://github.com/mantidproject/mantid/blob/master/Framework/API/inc/MantidAPI/IFunction.h) object from python directly via a thin wrapper python classes. `FitFunctionWrapper` will be the base class functionality common for all fit functions. Composite functions (`CompositeFunction`, `ProductFunction`, `Convolution`, `MultiDomainFunction`) will have their own wrappers inheriting from `FitFunctionWrapper`. If a fit function exibits a special behaviour a custom wrapper can be defined.
```

Comment:
Explicitly mention that the fit functions will appear in `simpleapi.fitfunction` namespace